File: docs/examples/python/delete_logs.py
"""Delete logs using the DELETE endpoint."""

import requests

BASE_URL = "http://localhost:3000/api"
TOKEN = "<token>"
PROJECT_UUID = "<project-uuid>"


def delete_logs() -> None:
    params = {"level": "DEBUG"}
    response = requests.delete(
        f"{BASE_URL}/logs/{PROJECT_UUID}",
        params=params,
        headers={"Authorization": f"Bearer {TOKEN}"},
        timeout=10,
    )
    response.raise_for_status()
    print("Deleted logs:", response.json()["deleted"])


if __name__ == "__main__":
    delete_logs()

File: docs/examples/typescript/deleteLogs.ts
import { LoggerApiClient } from './loggerClient';

/**
 * Deletes logs in bulk using a filter similar to the query parameters of the GET /logs endpoint.
 */
async function main(): Promise<void> {
  const client = new LoggerApiClient({ baseUrl: 'http://localhost:3000/api', token: '<token>' });

  try {
    const result = await client.deleteLogs('<project-uuid>', {
      level: 'DEBUG',
      endDate: '2024-05-01T00:00:00.000Z'
    });

    console.log('Deleted logs count:', result.deleted);
  } catch (error) {
    console.error('Failed to delete logs:', error);
  }
}

void main();
